### PR TITLE
PDA-17 Added endpoints to submit a new process form

### PR DESCRIPTION
**Pull request by ffleandro**
Also adds support to SubForm field types;

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/main/java/org/entando/plugins/pda/core/model/form/Form.java [hunk 1]
```diff
@@ -3,12 +3,26 @@
...
 
     private String id;
     private String name;
+    private String type;
+
+    @Singular
     private List<FormField> fields;
+
+    public FormField getFieldByName(String id) {
+        for (FormField field : fields) {
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/core/model/form/Form.java]: Can be replaced by:
```
return fields.stream()
                .filter(e -> id.equalsIgnoreCase(e.getId()))
                .findFirst().orElse(null);
```

**Review by werbth — COMMENTED** [review]
@ffleandro There is a comment here, but you can treat it as a suggestion.

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Pull request merged**